modules/standard/Math.chpl

```diff
@@ -1057,6 +1057,96 @@ module Math {
+  extern "j0f" proc j0(x: real(32)): real(32);
+
+  /* Returns the Bessel function of the first kind of order `0` of `x`. */
+  inline proc j0(x: real(64)): real(64) {
+    extern proc j0(x: real(64)): real(64);
+    return j0(x);
+  }
+
+
+  /* Returns the Bessel function of the first kind of order `1` of `x`. */
+  extern "j1f" proc j1(x: real(32)): real(32);
+
+  /* Returns the Bessel function of the first kind of order `1` of `x`. */
+  inline proc j1(x: real(64)): real(64) {
+    extern proc j1(x: real(64)): real(64);
+    return j1(x);
+  }
+
+
+  /* Returns the Bessel function of the first kind of order `n` of `x`. */
+  extern "jnf" proc jn(n: int, x: real(32)): real(32);
+
+  /* Returns the Bessel function of the first kind of order `n` of `x`. */
+  inline proc jn(n: int, x: real(64)): real(64) {
+    extern proc jn(n: c_int, x: real(64)): real(64);
+    return jn(n.safeCast(c_int), x);
+  }
+
+
+  /* Returns the Bessel function of the second kind of order `0` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  extern "y0f" proc y0(x: real(32), boundChecking = true): real(32);
+
+  /* Returns the Bessel function of the second kind of order `0` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  inline proc y0(x: real(64), boundChecking = true): real(64) {
+    if boundChecking {
+      if x < 0 then
+        halt("Input value for Bessel function of second kind must be greater than 0");
+    }
+    extern proc y0(x: real(64)): real(64);
+    return y0(x);
+  }
+
+
+  /* Returns the Bessel function of the second kind of order `1` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  extern "y1f" proc y1(x: real(32), boundChecking = true): real(32);
+
+  /* Returns the Bessel function of the second kind of order `1` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  inline proc y1(x: real(64), boundChecking = true): real(64) {
+    if boundChecking {
+      if x < 0 then
+        halt("Input value for Bessel function of second kind must be greater than 0");
+    }
+    extern proc y1(x: real(64)): real(64);
+    return y1(x);
+  }
+
+
+  /* Returns the Bessel function of the second kind of order `n` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  extern "y1f" proc yn(n: int, x: real(32), boundChecking = true): real(32);
```

Comment:
This should look like the yn below. You can't use the `extern "y1f" proc yn` syntax here because we need the Chapel function to convert from int to c_int.

Reply:
Oh sorry, just copied above lines, bad  in such names.

Reply:
I want to ask a thing, proc yn is the wrapper function to yn <which belongs to C stdlib> for the user, the user will call yn procedure, giving n as input, this n will be integer wrt to chapel, then this wrapper will call C function, for that C function the arguments needed will be a C type integer which is a 32 bit int (in contrast to 64 bit integer of chapel), so the user is giving integer it should be int not c_int, and it is casted to C type integer at the time of calling, so I left it there in my sense, because at that time I didn't get the need to take c_int, because caller is chapel, chapel will provide the simple integers.

Reply:
@mppf @ben-albrecht